rbm_analysis.py

```diff
@@ -0,0 +1,669 @@
+    of the RBM Markov fields
+
+    :param field: the RBM state (visible or hidden), numpy
+
+    :param sign: sign of the conversion 
+
+    :return: the Ising field
+
+    """
+    sign_field = np.full(field.shape, sign)
+
+    return (2.0 * field + sign_field).astype(int)
+
+
+def ising_magnetization(field):
+    #axis=1 to return the average field for each state dimension N_concsamp x 1 
+    m = np.abs((field).mean(axis=1))
+    return np.array([m, m * m])
+
+
+def energy(field):
+    N = parameters['ising']['size']
+    state = np.array(field).reshape((N, N))
+    E = np.sum((-state*np.roll(state, -1, axis=0) - state*np.roll(state, -1, axis=1)))/float(N*N)
+    #E = 0
+    #for i in range (0,N):
+    #  for j in range (0,N):
+    #    E += -field[N*i+j]*field[N*i+(j+1)%N] - field[N*i+j]*field[N*(i+1)%N+j]
+    #  E=E/float(N*N)
+    return np.asarray([E, E*E])
+
+
+def energy_concurrent_sampling(field, beta=1.0):
+    field = np.asarray(field)
+    return(beta*np.apply_along_axis(energy, 1, field).transpose())
+
+
+def ising_averages(mag_history, en_history, model_size, label=""):
+    """ 
+    Get observables using statistic of more idependent gibbs sampling.
+    
+    """
+    
+    resample_size = parameters['steps']-parameters['thermalization']      # number of states in each resampled set s_i
+    # using concurrent sampling
+    # magnetization
+    mag_matrix = mag_history[:, 0, :]        # get a matrix with just the magnetization, along the columns we have mag of different gibbs sampled states, along the lines differen conc samplings
+    mag_gibbs_avg = mag_matrix.mean(axis=0)  # take the mean across gibbs sampled states
+    mag = mag_gibbs_avg.mean()               # take the mean across concurrent sampled states
+    mag_error = mag_gibbs_avg.std()          # take std across concurrent sampled states
+
+    # susceptibility
+    susc_gibbs_avg = model_size*(mag_history[:, 1, :].mean(axis=0) - mag_gibbs_avg*mag_gibbs_avg)/parameters['temperature']
+    susc = susc_gibbs_avg.mean()             # take mean cross concurrent samplings
+    susc_error = susc_gibbs_avg.std()        # take std across concurrent sampled states
+
+    # energy
+    en_matrix = en_history[:, 0, :]
+    en_gibbs_avg = en_matrix.mean(axis=0)
+    en = en_gibbs_avg.mean()
+    en_error = en_gibbs_avg.std()
+
+    # heat capacity
+    cv_gibbs_avg = model_size*(en_history[:, 1, :].mean(axis=0) - en_gibbs_avg*en_gibbs_avg)/(parameters['temperature']*parameters['temperature'])
+    cv = cv_gibbs_avg.mean()
+    cv_error = cv_gibbs_avg.std()
+    
+    return mag, mag_error, en, en_error, susc, susc_error, cv, cv_error
+
+
+def ising_averages_gs(mag_history, en_history, model_size, label=""):
+    """ 
+    Get observables using statistic of a single gibbs sampling.
+    In order to get the same statistic as metropolis, in the json file use:
+    - steps = 2000000
+    - save interval = 1000
+    - thermalization = 20000
```

Comment:
actually you have to factor in thermalization if this behaves like rbm_sample in other words if you want 20000 points then you need:
steps = 2020001
save interval = 100
thermalization = 20000
this will get you exactly 20000 points